lib/import.js

```diff
@@ -0,0 +1,157 @@
+    }
+  }
+
+  return newIds
+}
+
+const main = async () => {
+  program
+    .version( '0.0.1' )
+    .arguments( '<input> <tableName>' )
+    .option( '-o --order-by <column>', 'order column name' )
+    .option( '-s --shabad-id <column>', 'Shabad ID column name' )
+    .option( '-S --source <column>', 'Source column name', -1 )
+    .option( '-2 --sttm-id <column>', 'STTM2 Shabad ID column name' )
+    .option( '-t, --translation <column>', 'column name of a translation', ( val, prev ) => [ ...prev, val ], [] )
+    .option( '-p, --page <column>', 'Page number column name' )
+    .option( '-l, --line <column>', 'Line number column name' )
+    .option( '-g, --gurmukhi <column>', 'Gurmukhi column name' )
+    .parse( process.argv )
+
+  const [ filename, tableName ] = program.args
+
+  if ( !( filename && tableName ) ) program.outputHelp()
+
+  // Extract the columnar options
+  const { orderBy, shabadId, sttmId, source, gurmukhi, translation, line, page } = Object
+    .entries( program.opts() )
+    .filter( ( [ , value ] ) => !!value )
+    .filter( ( [ key ] ) => columnOpts.includes( key ) )
+    .reduce( ( opts, [ key, value ] ) => ( { ...opts, [ key ]: value } ), {} )
+
+  console.log( `Running import CLI with options: ${JSON.stringify( program.opts() )}\n`.subheader )
+
+  // Open target DB and fetch all the target data, then close it
+  console.log( `Opening database ${filename}`.header )
+  const db = await sqlite.open( filename )
+  const lines = await db.all( `SELECT * FROM ${tableName} ORDER BY ${orderBy}` )
+  const shabadIds = ( await db.all( `SELECT DISTINCT ${shabadId} FROM ${tableName}` ) ).map( x => x[ shabadId ] )
+  const sources = ( await db.all( `SELECT DISTINCT ${source} FROM ${tableName}` ) ).map( x => x[ source ] )
+  await db.close()
+
+  // Generate some Line and Shabad IDs and map them in
+  console.log( 'Generating IDs'.header )
+  const newShabadIDs = await findIds( Shabads, 3, shabadIds.length )
+  const newLineIDs = await findIds( Lines, 4, lines.length )
+  const newSourceId = ( await Sources.query().orderBy( 'id', 'desc' ).first() ).id + 1
+  const newTranslationSourceId = ( await TranslationSources.query().orderBy( 'id', 'desc' ).first() ).id + 1
+
+  // Get last order IDs
+  const { orderId: shabadOrderId } = await Shabads.query().orderBy( 'order_id', 'desc' ).first()
+  const { orderId: lineOrderId } = await Lines.query().orderBy( 'order_id', 'desc' ).first()
+
+  console.log( 'Mapping data'.header )
+  // Map all the old shabad IDs to new ones
+  const shabadIDMap = shabadIds.reduce( ( ids, id, index ) => ( {
+    ...ids,
+    [ id ]: newShabadIDs[ index ],
+  } ), {} )
+
+  // Map sources to original shabad id
+  const sourceMap = lines.reduce( ( sourceMap, line ) => ( {
+    ...sourceMap,
+    [ line[ shabadId ] ]: sources.find( source => source === line[ source ] ) || -1,
+  } ), {} )
+
+  // Generate sources
+  const newSources = sources.map( ( name, index ) => ( {
+    name_english: `${name}-new`,
```

Comment:
Is `name` user-generated through the sqlite file somehow or is it via `-S` parameter? Why append `-new`? Please explain this a little more

Reply:
Via whatever the column that the `-S` has, but the reason `-new` is appended is to make it clear that you need to check it yourself and so that it doesn't clash with any existing sources of the same name during import